metrix/src/models.rs

```diff
@@ -1,7 +1,21 @@
-#[derive(Queryable)]
+use super::schema::metrics;
+use serde_json;
+use std::time::SystemTime;
+
+#[derive(Serialize, Deserialize, Queryable)]
+// #[table_name="metrics"] -- idk why it throws compile error
+// "error: cannot find attribute `table_name` in this scope" ...
 pub struct Metric {
     pub id: i32,
     pub metric_name: String,
-    pub body: String,
-    pub created_at: String,
+    pub data: serde_json::Value,
+    pub created_at: SystemTime,
+    pub updated_at: SystemTime,
+}
+
+#[derive(Serialize, Deserialize, Insertable)]
+#[table_name="metrics"]
+pub struct NewMetric {
```

Comment:
So it's got this incoming and outgoing model thing going on? Is that the only option? Can't have one model for doing both?

Reply:
I initially tried optionals, like in this tutorial: https://github.com/sean3z/rocket-diesel-rest-api-example/blob/master/src/hero.rs (which has been very useful, but some APIs are outdated now) but that didn't work, I kept getting compiler errors from the `diesel`'s Query generation system.

I came across this pattern in the diesel getting-started examples, I recalled it and did a swap and seems to work. There's still a lot I am still figuring out (clearly) so the retro-justification I am coming up with is that the schema for creation, is different than the schema for reads/updates, since some things are auto-generated by the backend. But yea, anyways...yes but no :D